src/use-swr.ts

```diff
@@ -302,27 +302,30 @@ function useSWR<Data = any, Error = any>(
   // display the data label in the React DevTools next to SWR hooks
   useDebugValue(stateRef.current.data)
 
-  const rerender = useState(null)[1]
-  let dispatch = useCallback((payload: actionType<Data, Error>) => {
-    let shouldUpdateState = false
-    for (let k in payload) {
-      if (stateRef.current[k] === payload[k]) {
-        continue
-      }
+  const [, rerender] = useState(null)
+  let dispatch = useCallback(
+    (payload: actionType<Data, Error>) => {
+      let shouldUpdateState = false
+      for (let k in payload) {
+        if (stateRef.current[k] === payload[k]) {
+          continue
+        }
 
-      stateRef.current[k] = payload[k]
-      if (stateDependencies.current[k]) {
-        shouldUpdateState = true
+        stateRef.current[k] = payload[k]
+        if (stateDependencies.current[k]) {
+          shouldUpdateState = true
+        }
       }
-    }
 
-    if (shouldUpdateState || config.suspense) {
-      // if component is unmounted, should skip rerender
-      // if component is not mounted, should skip rerender
-      if (unmountedRef.current || !initialMountedRef.current) return
-      rerender({})
-    }
-  }, [])
+      if (shouldUpdateState || config.suspense) {
+        // if component is unmounted, should skip rerender
+        // if component is not mounted, should skip rerender
+        if (unmountedRef.current || !initialMountedRef.current) return
+        rerender({})
+      }
+    },
+    [config.suspense]
```

Comment:
We need to ignore this dependence because we don't allow the `suspense` option to change during the lifecycle, it has to be static (otherwise it would be difficult to track the internal state).

Maybe we should warn the user when they try to change the option.

Reply:
@shuding Thank you for your feedback.  I've disabled the warning by the eslint-disable comment instead of adding it to the deps array.

> Maybe we should warn the user when they try to change the option.

That's a good idea. I think the warning isn't required on a production environment, but `swr` doesn't have a way to print warnings only for a development build. I'll propose it later.